graphistry/compute/gfql_unified.py

```diff
@@ -86,40 +180,76 @@ def gfql(self: Plottable,
         # Dict → DAG execution (convenience)
         g.gfql({'people': n({'type': 'person'})})
     """
-    # Handle dict convenience first (convert to ASTLet)
-    if isinstance(query, dict):
-        # Auto-wrap ASTNode and ASTEdge values in Chain for GraphOperation compatibility
-        wrapped_dict = {}
-        for key, value in query.items():
-            if isinstance(value, (ASTNode, ASTEdge)):
-                logger.debug(f'Auto-wrapping {type(value).__name__} in Chain for dict key "{key}"')
-                wrapped_dict[key] = Chain([value])
-            else:
-                wrapped_dict[key] = value
-        query = ASTLet(wrapped_dict)  # type: ignore
-    
-    # Dispatch based on type - check specific types before generic
-    if isinstance(query, ASTLet):
-        logger.debug('GFQL executing as DAG')
-        return chain_let_impl(self, query, engine, output)
-    elif isinstance(query, Chain):
-        logger.debug('GFQL executing as Chain')
-        if output is not None:
-            logger.warning('output parameter ignored for chain queries')
-        return chain_impl(self, query.chain, engine)
-    elif isinstance(query, ASTObject):
-        # Single ASTObject -> execute as single-item chain
-        logger.debug('GFQL executing single ASTObject as chain')
-        if output is not None:
-            logger.warning('output parameter ignored for chain queries')
-        return chain_impl(self, [query], engine)
-    elif isinstance(query, list):
-        logger.debug('GFQL executing list as chain')
-        if output is not None:
-            logger.warning('output parameter ignored for chain queries')
-        return chain_impl(self, query, engine)
-    else:
-        raise TypeError(
-            f"Query must be ASTObject, List[ASTObject], Chain, ASTLet, or dict. "
-            f"Got {type(query).__name__}"
-        )
+    # Recursion prevention - check if we're already in a policy execution
+    policy_depth = getattr(_thread_local, 'policy_depth', 0)
+    if policy and policy_depth >= 1:
+        logger.debug('Policy disabled due to recursion depth limit (depth=%d)', policy_depth)
+        policy = None  # Disable policy for recursive calls
+
+    # Set depth for this execution
+    if policy:
+        _thread_local.policy_depth = policy_depth + 1
+
+    try:
```

Comment:
breaking too much git history, make this call an internal (so no indent change) called by a parent who has the new try/catch